<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro">

  <gazebo>
    <plugin name="ros_control" filename="libgazebo_ros_control.so">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
    


<!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
     	
  </gazebo>
  

<!--  <gazebo>-->
<!--  	<plugin name="gazebo_grasp_fix" filename="libgazebo_grasp_fix.so">-->
<!--  	  <arm>-->
<!--  	  	<arm_name>ur5</arm_name>-->
<!--  	  	<palm_link>ee_link</palm_link>-->
<!--  	  	<gripper_link>vacuum_gripper0</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper1</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper2</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper3</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper4</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper5</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper6</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper7</gripper_link>-->
<!--  	  	<gripper_link>vacuum_gripper8</gripper_link>-->
<!--  	  </arm>-->
<!--  	  <forces_angle_tolerance>100</forces_angle_tolerance>-->
<!--  	  <update_rate>10</update_rate>-->
<!--  	  <grip_count_threshold>3</grip_count_threshold>-->
<!--  	  <max_grip_count>10</max_grip_count>-->
<!--  	  <release_tolerance>0.001</release_tolerance>-->
<!--  	  <disable_collisions_on_attach>false</disable_collisions_on_attach>-->
<!--  	  <contact_topic>__default_topic__</contact_topic>-->
<!--  	</plugin>-->
<!--  </gazebo>-->





  <!-- camera_link -->
  <gazebo reference="camera_link">
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <material>Gazebo/Green</material>
  </gazebo>

 <!-- camera -->
  <gazebo reference="camera_link">
    <sensor type="camera" name="camera1">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.007</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>ur5/usbcam</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera_link_optical</frameName>
        <!-- setting hackBaseline to anything but 0.0 will cause a misalignment
            between the gazebo sensor image and the frame it is supposed to
            be attached to -->
        <hackBaseline>0.0</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
        <CxPrime>0</CxPrime>
        <Cx>0.0</Cx>
        <Cy>0.0</Cy>
        <focalLength>0.0</focalLength>
      </plugin>
    </sensor>
  </gazebo>


</robot>
